### DISCUSSION TITLE
Apicurio Registry VSCode plugin

### DISCUSSION THREAD

**Discussion by jetmartin** [Apicurio/apicurio-registry#3974] [Show and tell]
Hi,

I'm not sure it's the right place to put it but I let you tell me.
I've made an Apicurio Registry [VSCode plugin](https://github.com/jetmartin/apicurio) some time ago to support my dev team in one of my previous project.

It's available on [VSC marketplace](https://marketplace.visualstudio.com/items?itemName=jetmartin.apicurio) as well as [OpenVSX](https://open-vsx.org/extension/jetmartin/apicurio).
It's not the best code ever and there are many improvement or additional features who can be managed but it was well appreciated by my team ans there are some downloads on OpenVSX and Marketplace.

Due to my actual activities I do have lack of time to be reactive in case of defect or to properly manage Apicurio lifecycle for plugin updates.
So I assume the best place for it would have been on the Apicurio organisation rather that on my personal account if you think it make sens.

Let me knew you'r thought about it.

Best.

J-Et. Martin

**Comment by EricWittmann** [comment]
I've only had time to look at the Marketplace page for it, but the demo videos look super cool!  I am going to have to send our legal team after you for unauthorized use of the logo, however!  (kidding)

Totally agree that this sort of thing would be great to have in the Apicurio org.

We'll discuss internally whether we have capacity to take this into the community.

Thanks!

**Reply by EricWittmann** [reply]
So after discussing it on the team, everything thinks it's a great idea to pull this work into the Apicurio github organization.  Thanks for the great work and the offer to contribute it back!

Normally I would suggest a transfer of the repository into the Apicurio GH org.  However, we are moving more in the direction of mono-repos for Apicurio projects.  So I think it probably makes more sense for it to be a new folder within the `apicurio-registry` repository (e.g. a new `vscode-plugin` folder).  In that case, we could simply copy the code.  But doing that would lose your attributions, unless you were the one to open the PR.

So what do you think?  If you don't care about the attribution, I can just clone your repo and copy all the code into the registry repo (with your permission).  Or instead you could open a PR doing the same.

(my preference would be the latter so that you retain the credit, but I'm happy to do either)

**Reply by jetmartin** [reply]
Hi @EricWittmann ,

Sorry for the lack of reactivity. Happy that you like that work !

I like the mono-repos approach, but...
Let me just check if it can cause issues for the VscMarketplace or openVSX publication as they grab some files by default...
If it'f fine, I will make a PR and update my repo to redirect users to the "official" Apicurio repo otherwise i'll came back to you with another idea.

I will also have to create an Apicurio ID on both stores, let's chat in private how to manage the secrets across the Apicurio org team.

If you will I can stay contributor in the Apicurio org to maintain the plugin.
Just, as you've just seen, I'm not the most responsive guy on Github (opensource is manage on my personal free time which I lake of....)
I will probably look at that over the WE or late next week.

Best.

**Reply by EricWittmann** [reply]
Sounds great, thanks!  I'll respond re: secrets via email.  :)

Also don't worry at all about lack of reactivity.  My job at Red Hat is full-time open source development and I'm **still** not very responsive at times!  lol
